lib/test/karma.js

```diff
@@ -14,60 +17,115 @@ function quote(value) {
+    //aggregate and append to karma.conf.js in the project root folder
+    gulp
+      .src(path.resolve(configFileName))
+      .on('data', function(karmaConfigFile) {
+        var contents = karmaConfigFile.contents.toString();
+        contents = contents.replace(filesAppendRegex, filesAppend);
+        karmaConfigFile.contents = new Buffer(contents);
+        stream.push(karmaConfigFile);
+      })
+      .on('end', function() {
         done();
       });
-    } else {
+  }
+
+  return through.obj(transformFn, flushFn);
+}
+
+/**
+ * Run karma once only with the karma config file present in the input stream.
+ * No output. Ends when the Karma process ends.
+ * Runs karma in a child process, to avoid `process.exit()` called by karma.
+ *
+ * @param {number} [bannerWidth] The width of banner comment, zero or omitted for none
+ * @returns {stream.Through} A through strconcateam that performs the operation of a gulp stream
+ */
+function karmaRun(bannerWidth) {
+  var options = {
+    configFile: undefined
+  };
+
+  return through.obj(function transformFn(file, encoding, done) {
+    options.configFile = file.path;
+    done();
+  },
+  function streamFn(done) {
+    var appPath = path.join(__dirname, 'karma-background.js');
+    var data    = querystring.escape(JSON.stringify(options));
+    var command = [ 'node', quote(appPath), data ].join(' ');
+
+    //TODO @bguiz replace reporter function with a standard karma reporter,
+    //and extract it to own module
+    //perhaps extend the spec reporter to do what is being done here, instead of post processing its output here
+    //check if there is a webstorm/ teamcity reporter
+    childProcess.exec(command, { cwd: process.cwd() }, function reporter(stderr, stdout) {
+      var report;
+      if (stdout) {
+        console.log(stdout);
+        report = stdout
+          .replace(/^\s+/gm, '')                                  // remove leading whitespace
+          .replace(/^(LOG.*\n$)/gm, options.silent ? '' : '$1')   // remove logging
+          .replace(/at\s+null\..*/gm, '')                         // remove file reference
+          .replace(/\n\n/gm, '\n')                                // consolidate consecutive line breaks
+          .replace(/^\n|\n$/g, '');                               // remove leading and trailing line breaks overall
+      } else if (stderr) {
+        console.log(stderr);
+
+        var analysis = /$Error\:\s*(.*)$/mg.exec(stderr);
+        report = analysis ? analysis[1] : stderr;
+      }
+      if (report) {
+        var width = Number(bannerWidth) || 0;
+        var hr    = new Array(width + 1);   // this is a good trick to repeat a character N times
+        var start = (width > 0) ? (hr.join('\u25BC') + '\n') : '';
+        var stop  = (width > 0) ? (hr.join('\u25B2') + '\n') : '';
+        process.stdout.write(start + '\n' + report + '\n\n' + stop);
+      }
       done();
-    }
+    });
   });
 };
+
+module.exports = {
+  createConfig: karmaConfig,
```

Comment:
I believe that it is misleading to change the mapping like this, at least when there is no good reason to not name the internal function the same as the export.
Check with @impaler.